server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/SqlPlanTask.java

```diff
@@ -0,0 +1,433 @@
+        this.result = new SqlPlanTaskResult();
+        this.parameters = JobUtils.fromJson(getJobParameters().get(JobParametersKeyConstants.META_TASK_PARAMETER_JSON),
+                SqlPlanJobReq.class);
+        this.connectionSession = generateSession();
+        this.executor = connectionSession.getSyncJdbcExecutor(ConnectionSessionConstants.CONSOLE_DS_KEY);
+        initFile();
+    }
+
+    @Override
+    protected boolean doStart(JobContext context) throws Exception {
+        String sqlContent = parameters.getSqlContent();
+        log.info("sql content={}", sqlContent);
+        // todo: Download the attachment that needs to execute SQL from OSS
+        SqlCommentProcessor processor = ConnectionSessionUtil.getSqlCommentProcessor(connectionSession);
+
+        List<OffsetString> offsetStringList = processor.split(new StringBuffer(), sqlContent);
+        result.setTotalStatements(offsetStringList.size());
+
+        for (int i = 0; i <= offsetStringList.size() - 1; i++) {
+            OffsetString offsetString = offsetStringList.get(i);
+            // The retry statement will write the result into the buffer, while executing a new SQL command will
+            // clear the buffer.
+            queryResultSetBuffer.clear();
+            String sql = offsetString.getStr();
+            try {
+                boolean success = retryStatement(sql);
+                // write all result into json file
+                appendResultToJsonFile(i == 0, i == offsetStringList.size() - 1);
+                // write result rows into csv file
+                writeCsvFiles(i + 1);
+                if (success) {
+                    result.incrementSucceedStatements();
+                } else {
+                    result.incrementFailedStatements();
+                    // only write failed record into error records file
+                    addErrorRecordsToFile(i + 1, sql);
+                }
+            } catch (Exception e) {
+                result.incrementFailedStatements();
+                addErrorRecordsToFile(i + 1, sql);
+                if (parameters.getErrorStrategy() == TaskErrorStrategy.ABORT) {
+                    canceled = true;
+                    break;
+                }
+                log.warn("Sql task execution failed, will continue to execute next statement.", e);
+            } finally {
+                result.incrementFinishedStatements();
+                log.info("finished statement, total={}, finished={}",
+                        result.getTotalStatements(), result.getFinishedStatements());
+            }
+        }
+        log.info("The sql plan task execute finished,result={}", result);
+
+        // all sql execute csv file list write to zip file
+        writeZipFile();
+        // upload file to OSS, also contains error record where is non-null
+        upload();
+        return true;
+    }
+
+
+
+    @Override
+    protected void doStop() {
+        tryExpireConnectionSession();
```

Comment:
`doClose` will be called after `doStop` in the task framework. So `tryExpireConnectionSession` may be duplicated.

Reply:
The tryExpireConnectionSession method will ascertain whether the connection session has been closed; if it remains open, the connection session will be expired.

Reply:
Modified as suggested, doClose will eventually be executed after doStop.